File: paper_planes-model/src/main/java/fr/sopra/model/Etape.java
package fr.sopra.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Positive;
import javax.validation.constraints.Size;

@Entity
@Table(name = "etape")

public class Etape {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ETA_ID")
	private int id;

	@Column(name = "ETA_DESCRIPTIF", columnDefinition = "TEXT", nullable = false)
	@NotEmpty
	private String descriptif;

	@Column(name = "ETA_IMAGE")
	@Size(max = 500)
	private String image;

	@Column(name = "ETA_VIDEO")
	@Size(max = 500)
	private String video;

	@Column(name = "ETA_ORDRE", nullable = false)
	@Positive
	private int ordre;

	@ManyToOne
	@JoinColumn (name="ETA_ORIGAMI_ID")
	private Origami origami;

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getDescriptif() {
		return descriptif;
	}

	public void setDescriptif(String descriptif) {
		this.descriptif = descriptif;
	}

	public String getImage() {
		return image;
	}

	public void setImage(String image) {
		this.image = image;
	}

	public String getVideo() {
		return video;
	}

	public void setVideo(String video) {
		this.video = video;
	}

	public int getOrdre() {
		return ordre;
	}

	public void setOrdre(int ordre) {
		this.ordre = ordre;
	}

	public Origami getOrigami() {
		return origami;
	}

	public void setOrigami(Origami origami) {
		this.origami = origami;
	}

	
}

File: paper-planes-angular/app/etape.ts
export class etape {
  private ordre: number;
  private descriptif: string;
  private image: string;
  private video: string;
  private href: string;
<<<<<<< HEAD

=======
>>>>>>> angular
  constructor(private json?: any) {
      if (json !== undefined) {
          this.ordre = json.ordre;
          this.descriptif = json.descriptif;
          this.image = json.image;
          this.video = json.video;
<<<<<<< HEAD

=======
>>>>>>> angular
          if (json._links !== undefined) {
              this.href = json._links.self.href;
          }
      }
  }
<<<<<<< HEAD

  public getOrdre(): number {
      return this.ordre;
  }

public getDescriptif(): string {
      return this.descriptif;
  }

  public getImage(): string {
        return this.image;
    }

    public getVideo(): string {
          return this.video;
      }

=======
  public getOrdre(): number {
      return this.ordre;
  }
public getDescriptif(): string {
      return this.descriptif;
  }
  public getImage(): string {
        return this.image;
    }
    public getVideo(): string {
          return this.video;
      }
>>>>>>> angular
  public getHref(): string {
      return this.href;
  }
}
